app/helpers/application_helper.rb

```diff
@@ -23,14 +23,12 @@ def sent_requests(id)
   end
 
   def current_user_requests(id)
-    FriendRequest.exists?(user_id: current_user.id, friend_id: id)
+    current_user.friend_requests.exists?(friend_id: id)
   end
 
   def pending_requests(id)
-    request = FriendRequest.find_by(user_id: current_user.id, friend_id: id)
-    request_two = FriendRequest.find_by(user_id: id, friend_id: current_user.id)
-
-    request.nil? ? request_two.confirmed : request.confirmed
+    request = current_user.friends.find_by(user_id: id)
+    request.nil? ? true : false
   end
 
   def pending_invitations
```

Comment:
```rb
# User model

# Users who needs to confirm friendship
has_many :pending_friendships, -> { where confirmed: false }, class_name: "FriendRequest", foreign_key: "user_id"
has_many :pending_friends, through: :pending_friendships, source: :friend
```

Reply:
We actually are not trying to get users who need confirmation. Our helper method is just to check if a user is already friend's with another user.